addons/account/static/src/js/reconciliation/reconciliation_model.js

```diff
@@ -1270,8 +1274,8 @@ var StatementModel = BasicModel.extend({
             'base_amount': amount,
             'percent': values.amount_type === "percentage" ? values.amount : null,
             'link': values.link,
-            'display': true,
-            'invalid': true,
+            'display': amount !== null,
+            'invalid': amount == null,
```

Comment:
This seems tricky and wrong for me, what will happen when amount is not null? invalid will be false but beforehand it was always true.. Also there is some places in the code where they change the value of invalid (_blurProposition) for example.
Also if the regex had a match but the matched element isn't a float, the line would be visible but with amount = NaN. So why don't we do something like this (which would be a one line fix and also prevent problem by tempering with display and invalid as they were always true before).
You could just change amount = null by amount = 0 and the parsefloat(matching) in the following:
```
case 'regex':
                var matching = line.st_line.name.match(new RegExp(values.amount_from_label_regex))
                amount = null; //-> amount = 0
                if (matching && matching.length == 2) {
                    matching = matching[1].replace(new RegExp('\\D' + values.decimal_separator, 'g'), '');
                    matching = matching.replace(values.decimal_separator, '.');
                    amount = parseFloat(matching); // -> parseFloat(matching) || 0;
                    amount = line.balance.amount > 0 ? amount : -amount;
                }
                break;
```
In both case a line would be displayed with 0 amount so the user would understand that the regex did not work, if you display nothing, no indication is displayed that the regex did not work and user might click several time on the button and then files a bug report.